utils/node_local_test.py

```diff
@@ -1884,10 +1900,107 @@ def test_rename(self):
         if dfuse.stop():
             self.fatal_errors = True
 
-        # Finally, perform some more I/O so we can tell from the dfuse logs where the test ends and
-        # dfuse teardown starts.  At this point file 1 and file 2 have been deleted.
-        time.sleep(1)
-        print(os.statvfs(self.dfuse.dir))
+    @needs_dfuse
+    def test_rename_existing(self):
+        """Test that rename clobbers files correctly"""
+
+        def _go(root):
+            # Create all three files in the dfuse instance we're checking.
+
+
+            dfd = os.open(root, os.O_RDONLY)
+            try:
+
+                # Test renaming a file into a directory.
+                pre_fname = os.path.join(root, 'file')
+                with open(pre_fname, 'w') as fd:
+                    fd.write('test')
+                dname = os.path.join(root, 'dir')
+                os.mkdir(dname)
+                post_fname = os.path.join(dname, 'file')
+                # os.rename and 'mv' have different semantics, use mv here which will put the file
+                # in the directory.
+                subprocess.run(['mv', pre_fname, dname], check=True)
+                self._check_dirs_equal(['file'], dname)
+
+                os.unlink(post_fname)
+                os.rmdir('dir', dir_fd=dfd)
+
+                # Test renaming a file over a directory.
+                pre_fname = os.path.join(root, 'file')
+                with open(pre_fname, 'w') as fd:
+                    fd.write('test')
+                dname = os.path.join(root, 'dir')
+                os.mkdir(dname)
+                post_fname = os.path.join(dname, 'file')
+                # Try os.rename here, which we expect to fail.
+                try:
+                    os.rename(pre_fname, dname)
+                    self.fail()
+                except IsADirectoryError:
+                    pass
+                os.unlink(pre_fname)
+                os.rmdir('dir', dir_fd=dfd)
+
+                # Check renaming a file over a file.
+                for index in range(2):
+                    with open(os.path.join(root, 'file.{}'.format(index)), 'w') as fd:
+                        fd.write('test')
+
+                print(os.listdir(dfd))
+                os.rename('file.0', 'file.1', src_dir_fd=dfd, dst_dir_fd=dfd)
+
+                self._check_dirs_equal(['file.1'], root)
+                os.unlink('file.1', dir_fd=dfd)
+
+                # dir onto file.
+                dname = os.path.join(root, 'dir')
+                os.mkdir(dname)
+                fname = os.path.join(root, 'file')
+                with open(fname, 'w') as fd:
+                    fd.write('test')
+                try:
+                    os.rename(dname, fname)
+                    self.fail()
```

Comment:
(pylint-no-value-for-parameter) No value for argument 'self' in staticmethod call